engine/card.py

```diff
@@ -22,9 +25,33 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    def make_public(self, information_type):
+    def matches(self, valid_colors=None, valid_numbers=None):
+        return self.does_color_match(valid_colors) and self.does_number_match(valid_numbers)
+
+    def does_color_match(self, valid_colors=None):
+        if isinstance(valid_colors, Card):
+            valid_colors = valid_colors.color
+        return not valid_colors or self.color.matches(valid_colors)
+
+    def does_number_match(self, valid_numbers=None):
+        if isinstance(valid_numbers, int):
+            valid_numbers = [valid_numbers]
+        return not valid_numbers or self.number in valid_numbers
+
+    def make_public(self, information_type, information=None):
         if information_type == 'color':
-            self.public_color = self.color  # Todo: Rainbow/wild color management
+            # No color information suggested or the `information` color does not match the `public_color`.
+            if (not information or
+                    # Ensure public color is not `None` or `False` before checking if it matches `information`.
+                    (self.public_color and not self.public_color.matches(information))):
```

Comment:
Sanity check for me:
If information is left as None, but someone's already noted self.public_color is 'b' then this let's you say "well since it's not being specified as 'b' then we set it to whatever it's actual color is".